src/components/checkbox/checkbox.js

```diff
@@ -171,12 +197,48 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
+        if (indeterminateStateEnabled) {
+          setIndeterminateState();
+          // The checkbox should never be in an indeterminate="true" AND checked state.
+          // To prevent this, if the checkbox is in an indetermiante
+          // state, we skip the code below.
+          return;
+        }
+
         if(ngModelCtrl.$viewValue) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminateState() {
+        var isIndeterminate = element.attr('indeterminate') === 'true' || false;
```

Comment:
Ahh so it does seem like `parseAttributeValue` does not exist. BUT `parseAttributeBoolean` does exist. Unfortunately because we can accept a function now as well as a string, we can't use `parseAttributeBoolean` :(